### PR TITLE
[complex32] add chalf alias for complex32 and chalf method

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: https://github.com/pytorch/pytorch/issues/74537

Adds chalf alias for complex32 and also adds method `chalf` similar to `cfloat, cdouble`

TODO:
* [x] Add docs
* [x] Add override

### CODE REVIEW

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -357,6 +357,9 @@
     SparseCPU, SparseCUDA: sgn_sparse_out
     SparseCsrCPU, SparseCsrCUDA: sgn_sparse_csr_out
 
+- func: chalf(Tensor self, *, MemoryFormat? memory_format=None) -> Tensor
```
Comment (anjali411) [inline aten/src/ATen/native/native_functions.yaml:360]: I don't think we want to make it an aten function

**Review by anjali411 — CHANGES_REQUESTED** [review]
don't see why chalf needs to be an aten function. 

also ref: https://github.com/pytorch/pytorch/pull/34093 this was the PR where we added aliasing for complex32, complex64

**Comment by kshitij12345** [comment]
Ping @anjali411

**Review by anjali411 — APPROVED** [review]

**Comment by anjali411** [comment]
let's merge after the CI is green (the current failures seem due to upstream breakage)

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -14490,6 +14490,21 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
                # RuntimeError: attribute lookup is not defined on builtin
                DecorateInfo(unittest.expectedFailure, 'TestJit', 'test_variant_consistency_jit'),
            )),
+    OpInfo('chalf',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:14493]: See https://github.com/pytorch/pytorch/pull/75958 for an updated pattern

Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:14493]: Thanks!

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -14490,6 +14490,31 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
+           op=lambda x, *args, **kwargs: x.chalf(*args, **kwargs),
+           dtypes=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
+           supports_out=False,
+           sample_inputs_func=sample_inputs_conversion,
+           skips=(
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:14498]: Had to add a lot of skips. PTAL :) Thanks! @anjali411

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -14490,6 +14490,31 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
+               # autograd tests don't handle operators that change dtype
```
Comment (anjali411) [inline torch/testing/_internal/common_methods_invocations.py:14499]: I think they can because they work for cfloat and cdouble for instance

**Review by anjali411 — APPROVED** [review]

**Pull request closed without merging**